### PR TITLE
fix(starfish): Remove http method if already on transaction

### PR DESCRIPTION
**Pull request by AbhiPrasad**
Some SDKs send `http.method` as part of the description. To prevent rendering duplicate methods (for ex. `GET GET /route`), this patch updates the body cell renderer to only set an http method if it has not been defined already.

### Before

image

### After

image

### CODE REVIEW

File: static/app/views/starfish/views/webServiceView/endpointList.tsx [hunk 1]
```diff
@@ -81,8 +81,9 @@ function EndpointList({eventView, location, organization, setError}: Props) {
 
     if (field === 'transaction') {
       let prefix = '';
-      if (dataRow['http.method']) {
-        prefix = `${dataRow['http.method']} `;
+      const method = dataRow['http.method'];
+      if (method && !(dataRow.transaction as string).startsWith(method as string)) {
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/webServiceView/endpointList.tsx:85]: I hate this casting but idk how else to do this 🤷

Comment (gggritso) [inline static/app/views/starfish/views/webServiceView/endpointList.tsx:85]:
```suggestion
      if (method && !dataRow.transaction.toString().startsWith(method as string)) {
```

**Review by gggritso — APPROVED** [review]
Not furious at the `as string` since the data type is actually known, but a more "correct" solution is to convert to the right type!

File: static/app/views/starfish/views/webServiceView/endpointList.tsx (hunk 1 above)
Comment (AbhiPrasad) [reply static/app/views/starfish/views/webServiceView/endpointList.tsx:85]: Good call!

**Pull request merged**